src/PortingAssistant.Client.Common/Utils/FileSystemAccess.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Security.Principal;
+using System.Security.AccessControl;
+
+namespace PortingAssistant.Client.Common.Utils
+{
+    public static class FileSystemAccess
+    {
+        /// <summary>
+        /// Checks directory and all content for write access
+        /// </summary>
+        /// <param name="path">Directory path</param>
+        /// <returns>List of items without write access</returns>
+        public static List<string> CheckWriteAccessForDirectory(string path)
+        {
+            var result = new List<string>();
+            foreach (string file in Directory.GetFiles(path))
```

Comment:
Are we checking all the files? Do we need to make the files list smaller? (for example, code files, project files, etc...)

Reply:
Good point. I couldn't think of any other file types that would be significant, so currently changed to check .cs and .csproj files.